invoke-ai/InvokeAI
# feat: write-only canvas metadata

## Summary

Store canvas objects in the metadata for both images generated by canvas graphs _and_ images created when clicking the `Save to Gallery` button.

Canvas objects (images, brush strokes, shapes, masks, etc) are stored as is. There is no recall capability - this is planned _only_ for a future canvas v2, based on the control layers renderer and state management.

## Related Issues / Discussions

I don't think we've had any requests for canvas metadata without the ability to recall, so this doesn't resolve any open issues.

## QA Instructions

- Generate on canvas. Every generation, in every mode - txt2img, img2img, inpaint, outpaint, sd1 & sdxl - should have a `_canvas_objects` property in the metadata.
- Click `Save to Gallery` on canvas after generating. Should have `_canvas_objects` property in the metadata.

## Merge Plan

n/a

## Checklist
